File: little-bee/controller/router/video.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_VIDEO = "/config/video"
const URL_ROUTER_VIDEO_BY_ID = "/config/video/:video-id"
const URL_ROUTER_VIDEO_QUERY = "/config/video/query"
const URL_ROUTER_VIDEO_ALL = "/config/video/all"

func VideoRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_VIDEO, config.CreateVideo)
	r.GET(URL_ROUTER_VIDEO_BY_ID, config.RetrieveVideo)
	r.PUT(URL_ROUTER_VIDEO, config.UpdateVideo)
	r.DELETE(URL_ROUTER_VIDEO_BY_ID, config.DeleteVideo)
	r.PUT(URL_ROUTER_VIDEO_BY_ID, config.DeleteVideoWithFlag)
	return r
}

File: frontend/console/src/api/video.js
import Axios from "@/http";

export function createVideo(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/video', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveVideo(videoId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/video/${ videoId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateVideo(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/video', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteVideo(videoId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/video/${ videoId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteVideoWithFlag(videoId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/video/${ videoId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllVideos() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/video/all').then(res => {
            resolve(res)
        })
    })
}

export function queryVideo (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/video/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}